src/EditorFeatures/TestUtilities/LanguageServer/LspMutatingTestWorkspace.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Roslyn.Test.Utilities
+{
+    internal sealed class LspMutatingTestWorkspace : TestWorkspace, IMutatingLspWorkspace
```

Comment:
exists just to have a separate impl that marks itself as the mutating form.  see above comment for alternative approach that might be cleaner.